MatrixSDK/Space/MXSpaceService.swift

```diff
@@ -162,71 +183,42 @@ public class MXSpaceService: NSObject {
-                    var parentIds = parentIdsPerRoomId[childSpace.spaceId] ?? Set<String>()
-                    parentIds.insert(space.spaceId)
-                    parentIdsPerRoomId[childSpace.spaceId] = parentIds
-                }
-            }
-            
-            let rootSpaces = output.spaces.filter { space in
-                return parentIdsPerRoomId[space.spaceId] == nil
-            }
-            
-            var ancestorsPerRoomId: [String: Set<String>] = [:]
-            var descendantsPerRoomId: [String: Set<String>] = [:]
-            rootSpaces.forEach { space in
-                self.buildRoomHierarchy(with: space, visitedSpaceIds: [], ancestorsPerRoomId: &ancestorsPerRoomId, descendantsPerRoomId: &descendantsPerRoomId)
-            }
-            
-            var orphanedRoomIds: Set<String> = Set<String>()
-            var orphanedDirectRoomIds: Set<String> = Set<String>()
-            for room in rooms {
-                if !room.isDirect && parentIdsPerRoomId[room.roomId] == nil {
-                    orphanedRoomIds.insert(room.roomId)
-                } else if room.isDirect && parentIdsPerRoomId[room.roomId] == nil {
-                    orphanedDirectRoomIds.insert(room.roomId)
-                }
+        var directRoomIds = Set<String>()
+        let roomIds: [String] = self.session.rooms.compactMap { room in
+            if room.isDirect {
+                directRoomIds.insert(room.roomId)
             }
-
-            self.graph = MXSpaceGraphData(
-                spaceRoomIds: output.spaces.map({ space in
-                    space.spaceId
-                }),
-                parentIdsPerRoomId: parentIdsPerRoomId,
-                ancestorsPerRoomId: ancestorsPerRoomId,
-                descendantsPerRoomId: descendantsPerRoomId,
-                rootSpaceIds: rootSpaces.map({ space in
-                    space.spaceId
-                }),
-                orphanedRoomIds: orphanedRoomIds,
-                orphanedDirectRoomIds: orphanedDirectRoomIds)
-            
-            if !(self.session.store?.store?(self.graph) ?? false) {
-                MXLog.error("[Spaces] buildGraph: failed to store space graph")
+            return room.roomId
+        }
+        
+        let output = PrepareDataResult()
+        MXLog.debug("[Spaces] preparing data for \(roomIds.count) rooms")
+        self.prepareData(with: roomIds, index: 0, output: output) { result in
+            while !output.computingSpaces.isEmpty {
+                sleep(1)
             }
             
-            // TODO improve updateNotificationsCount and call the method to all spaces once subspaces will be supported
-            self.notificationCounter.computeNotificationCount(for: rootSpaces, with: rooms, ancestorsPerRoomId: ancestorsPerRoomId)
-
-            MXLog.debug("[Spaces] buildGraph ended after \(Date().timeIntervalSince(startDate))s")
+            MXLog.debug("[Spaces] data prepared in \(Date().timeIntervalSince(startDate))")
```

Comment:
Can we use the common formatting for log, ie `MXLog.debug("[MXSpaceService] buildGraph: Data prepared ...")`?
Can you apply it everywhere, please? I just noticed it.